src/views/Update/components/UpdatePage/index.js

```diff
@@ -0,0 +1,365 @@
+        stringArr.forEach((word) => {
+          if (word === 'u.s.') return 'U.S.';
+        });
+        return stringArr.join(' ');
+      });
+      setCountries(allCountries);
+    });
+  }, []);
+
+  const currentInfo = useMemo(
+    () => ({
+      salutation: profile.Title
+        ? profile.Title.charAt(0).toUpperCase() + profile.Title.slice(1).toLowerCase()
+        : '',
+      firstName: profile.FirstName ?? '',
+      lastName: profile.LastName ?? '',
+      middleName: profile.MiddleName ?? '',
+      nickName: profile.NickName ?? '',
+      personalEmail: profile.PersonalEmail ?? '',
+      workEmail: profile.WorkEmail ?? '',
+      aEmail: profile.aEmail ?? '',
+      preferredEmail: profile.PreferredEmail ?? '',
+      doNotContact: profile.doNotContact ?? false,
+      doNotMail: profile.doNotMail ?? false,
+      homePhone: profile.HomePhone ?? '',
+      workPhone: profile.WorkPhone ?? '',
+      mobilePhone: profile.MobilePhone ?? '',
+      preferredPhone: profile.PreferredPhone ?? '',
+      //Homestreet lines are inverted in alumni SQL
+      address1: profile.HomeStreet2 ?? profile.HomeStreet1 ?? '',
+      address2: profile.HomeStreet2 && profile.HomeStreet1 ? profile.HomeStreet2 : '',
+      city: profile.HomeCity ?? '',
+      state: profile.HomeState ?? '',
+      zip: profile.HomePostalCode ?? '',
+      country: profile.HomeCountry ?? '',
+      married: profile.Married === 'Y' ? true : false,
+    }),
+    [profile],
+  );
+  const [updatedInfo, setUpdatedInfo] = useState(currentInfo);
+  const [openConfirmWindow, setOpenConfirmWindow] = useState(false);
+  const [isSaving, setSaving] = useState(false);
+  const [snackbar, setSnackbar] = useState({ message: '', severity: '', open: false });
+
+  const hasNoChange = useMemo(() => {
+    for (const field in currentInfo) {
+      if (currentInfo[field] !== updatedInfo[field]) {
+        return false;
+      }
+    }
+    return true;
+  }, [updatedInfo, currentInfo]);
+
+  // handles all field changes
+  const handleChange = (event) => {
+    const getNewInfo = (currentValue) => {
+      return {
+        ...currentValue,
+        [event.target.name]:
+          event.target.type === 'checkbox' ? event.target.checked : event.target.value,
+      };
+    };
+    setUpdatedInfo(getNewInfo);
+  };
+
+  const createSnackbar = (message, severity) => {
+    setSnackbar({ message: message, severity: severity, open: true });
+  };
+
+  // returns label given name of field
+  const getAssociatedLabel = (fieldName) => {
+    let label = '';
+    allFields.forEach((field) => {
+      if (fieldName === field.name) label = field.label;
+    });
+    return label;
+  };
+
+  /**
+   * @param updatedInfo updated information fields object
+   * @param currentInfo old/saved information fields object
+   * @returns {Array<Object>} Array of updated fields in a subobject
+   */
+  function getUpdatedFields(updatedInfo, currentInfo) {
+    var updatedFields = [];
```

Comment:
Do not use the `var` keyword. It is bad practice in modern JavaScript.
```suggestion
    const updatedFields = [];
```

Reply:
Yep I did this in the next commit (haven't pushed)